File: medium/LongestConsecutiveSequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 1)
            return 1;
        std::sort(nums.begin(), nums.end());
        int ans = 0;
        int temp = 1;
       for(int i = 1; i < nums.size(); i++){
           if(nums[i] - nums[i - 1] == 1)
                temp += 1;
            if(temp > ans)
                ans = temp;
            if(nums[i] - nums[i - 1] > 1)
                temp = 1;
       }
       return ans;
    }
};

File: medium/LongestConsecutiveSequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 1
        new = sorted(nums)
        ans = 0
        temp = 1
        for i in range(1, len(new)):
            if new[i] - new[i - 1] == 1:
                temp += 1
            if temp > ans:
                ans = temp 
            if new[i] - new[i - 1] > 1:
                temp = 1
        return ans

